### DISCUSSION TITLE
Clarify how to change the keymap on mispelled words

### DISCUSSION THREAD

**Discussion by jpggithub** [minad/jinx#190] [Q&A] [answered]
I use personal keybindings for move/delete words or char in org-mode and text-mode.

For example, I use `M-p` is bound to `backward-kill-word`.

`M-p` is bound with `jinx-previous` when jinx is activated and the point is on a mispelled word.

How can I preserve my initial keybinding?

I tried the both following configurations without success.

```
(add-hook 'global-jinx-mode-hook #'(lambda ()
				     (define-key jinx-overlay-map "M-p" 'beginning-of-line)))
```

```
(add-hook 'jinx-mode-hook #'(lambda ()
				     (define-key jinx-overlay-map "M-p" 'beginning-of-line)))
```

**Comment by minad** [comment, marked as answer]
Your usage of `define-key` is not correct - it does not accept a key string like `"M-p"`. I suggest to read the manual or the docstring of these functions. Also you could use `keymap-unset` or `keymap-set` instead, which are more user friendly.

**Reply by jpggithub** [reply]
Thank your for pointing out the syntax problem, I forgot the usual `(kbd "M-n")`. ouch!
Also thank you for the both functions that I didn't know.
I modified my init.el with:
```
(require 'jinx)
(add-hook 'global-jinx-mode-hook (keymap-unset jinx-overlay-map "M-n"))
(add-hook 'global-jinx-mode-hook (keymap-unset jinx-overlay-map "M-p"))
```
It works like a charm.
